src/main/groovy/com/github/erdi/gradle/webdriver/repository/DriverUrlsConfiguration.groovy

```diff
@@ -52,19 +55,21 @@ class DriverUrlsConfiguration {
             architectures << X86
         }
         def platform = platform(spec.os)
-        def bits = architectures.collect { bit(it) }
+        def bits = architectures.collect { bit(it) } as Set
+        def archs = architectures.collect { arch(it) } as Set
         def versionPattern = ~spec.version
 
         def matchingDrivers = drivers.findAll {
             it.name == spec.name &&
                 it.platform == platform &&
                 versionPattern.matcher(it.version.toString()).matches() &&
-                bits.contains(it.bit)
+                bits.contains(it.bit) &&
+                (it.arch == null || archs.contains(it.arch))
```

Comment:
I don't think the `it.arch == null` is quite right. I believe that we should only be matching download specs which do not specify architecture if architectures contains any of `X86` or `X86_64` so I will change it to 
```
it.arch == null && architectures.intersect([X86, X86_64])
```